### ISSUE TITLE
High cpu use

### ISSUE THREAD

**Issue by gerroon** [laurent22/joplin#995]
Hi

It seems like there is a new issue just sneaked in either in 115 or in 116.

Joplin seems to be using high cpu in the background while I am not interacting, like when it is in tray and it seems to be constant use. Since I am not  a dev I do not how to debug this. Sometimes this usage goes for a long time sometimes couple minutes but rather frequently. I do not remember this issue before

See a snapshot below of how it looks (107 is the cpu usage)

`
PID USER      PRI  NI  VIRT   RES   SHR S CPU% MEM%   TIME+  Command                                                                     
31689 USER     20   0 1059M  160M 64368 R 107.  2.0 10:07.67 /tmp/.mount_JoplindFzVZj/app/joplin --type=renderer --no-sandbox --primordia
`

`Joplin 1.0.116 (prod, linux) Appimage, Debian Buster x64`

**Comment by laurent22** [comment]
For me on Windows it's at 0% CPU when not in use. Is there something special about what's open when CPU usage is high? Like maybe a particularly large note, or with many images or Katex formulas? Does it do this with all the notes?

**Comment by gerroon** [comment]
All my notes are simple text files with some urls mostly.  

Here I am hearing my fan going nuts while I am replying to you. This issue was not resolved with a restart of Joplin.

**Comment by laurent22** [comment]
I think I know, on Linux I've recently made a change to keep the app awake while sync is in progress, so maybe that's what's making the CPU usage abnormally high. Does it stop doing it if sync is stopped?

**Comment by gerroon** [comment]
I am looking at it, it says sync completed but the cpu usage is still at 107, which is interesting, it seems like some threads get stuck at certain usage.

**Comment by laurent22** [comment]
I'll remove this hack in the next release. I'm clueless as to why the UI is sleeping on Linux while sync is in progress, but that fix I've added doesn't seem to be the right one.

**Comment by Jestre** [comment]
This seems to be a problem on MacOS as well.  I installed it this morning via Homebrew, and after configuring sync it pulled in my 20 or so notes.  I minimized it and about 20 minutes later I noticed the fan was running, so I checked Activity Monitor and noticed that Joplin Helper was at 110.9% CPU.

**Comment by CalebJohn** [comment]
@gerroon If the issue is that going back to an older version results in a messed up half light/half dark theme, you can just set the light theme before reverting back to an older version. If the issue is an attachment to the dark theme, then my apologies, I'm no help.

**Comment by laurent22** [comment]
What sync targets are you using? (Nextcloud, Dropbox, etc.)

**Comment by Jestre** [comment]
> What sync targets are you using? (Nextcloud, Dropbox, etc.)

Mine is Nextcloud

**Comment by laurent22** [comment]
I can't replicate this in either macOS nor Linux unfortunately, so more information will be needed. Any chance you could enable [debug mode](https://joplin.cozic.net/debugging/) and see if you're getting any error or warning, or perhaps, suddenly many log statements in either the log file or console?

**Comment by laurent22** [comment]
Does it happen right away when you start the app, or only after a certain time?

Does it happen only when the app is in the tray or all the time?

Are there any changes when you sync or when sync stops?

**Comment by gerroon** [comment]
That is all I have in the console and still using %107 cpu (set to verbose)

/tmp/.mount_JoplinIePbaP/app/resources/app/lib/logger.js:120 2018-11-21 14:01:35: "Starting Clipper server on port 41184"

In my case most of the time it seems to be using high cpu right away then repeatedly even if it goes regular levels after sometime

It happens all the time whether in tray or in background.

But there is a speculiar situation, it seems to start usingn high cpu when a note is not selected, ie "no notes in this category" message is avail, then if I select a note the cpu level drops right away.

**Comment by Jestre** [comment]
> But there is a peculiar situation, it seems to start usingn high cpu when a note is not selected, ie "no notes in this category" message is avail, then if I select a note the cpu level drops right away.

This seems to be the trigger:   If I create a new Notebook and select it (thus receiving the above message), the CPU immediately spikes, and the following message is written to `log.txt` approximately 320 times per second:

    2018-11-21 15:16:24: "Reducer action", "SET_NOTE_TAGS"

Thus

    $ grep '2018-11-21 15:16:24' log.txt | wc -l
    Wed Nov 21 15:16:43 2018
     321

**Comment by gerroon** [comment]
Any progress on this issue?

**Comment by laurent22** [comment]
@Abijeet, I have disabled the tag bar is the latest release as the CPU issue would need to be fixed. I've done so by adding a flag so that we can turn it back on once it's fixed.

**Comment by Abijeet** [comment]
@laurent22 - I'll check this out.
